### ISSUE TITLE
404 error on item view after first save - using static cache (half strategy) 

### ISSUE THREAD

**Issue by jamie-b5** [statamic/cms#9668]
### Bug description

We are using static caching. Half strategy.

When we add an item to a collection and hit save the front end item url throws a 404. It does this until we save the item a second time, at which point the url resolves to the correct record.

### How to reproduce

1. Enable static caching (half strategy)
2. Create a new record in a collection
3. Visit the url of the item (front end) - this shows a 404
4. Save the record a second time
5. Visit the url of the item - now works

### Environment

```yaml
Environment
Application Name: Dave Whyte Animation
Laravel Version: 9.52.16
PHP Version: 8.1.21
Composer Version: 2.4.2
Environment: production
Debug Mode: OFF
URL: davewhyteanimation.com
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 0
Antlers: runtime
Sites: 1
Stache Watcher: Disabled
Static Caching: half
Version: 4.52.0 PRO
```

### Installation

Other (please explain)

### Antlers Parser

None

### Additional details

Statamic install - cloned from a slightly earlier repo then updated.

**Comment by chopperoon** [comment]
A client has also reported the same issue to us regarding their site also using half strategy caching.

**Comment by morhi** [comment]
@jamie-b5 @CjRootStudio I have similar issue. Does the 404 error disappear when you save the entry and then warm or refresh the Stache instead of saving the entry a second time?

**Comment by jamie-b5** [comment]
This seemed to be resolved for us by upgrading the core Statamic version FYI.

**Comment by CjRootStudio** [comment]
Thanks Jamie - did you upgrade to version 5 or is there a version 4 release that fixed it for you?

**Comment by morhi** [comment]
I think https://github.com/statamic/cms/pull/10135 fixed it. Can you confirm that? It was released yesterday.

**Comment by jamie-b5** [comment]
I will upgrade Statamic and advise.

Many thanks.

**Comment by jamie-b5** [comment]
Upgraded to v5.49.1.

Error still occurs. It also does seem to be tied to the order/max depth setting of the collection as pointed out by [CjRootStudio](https://github.com/CjRootStudio).

I created a page (collection does not have ordering) and this did not 404 after the first save.

I created a vehicle (collection has ordering enabled, max depth = 1) this DID 404 after the first save.

I edited the vehicles collection and removed the ordering. I subsequently added a new test vehicle - this did not 404 after the first save.

Shall I raise a new bug report issue? Or will you re-open this one?
